### PR TITLE
Hosting Configuration: Update Learn More Links

### PR DESCRIPTION
**Pull request by katinthehatsite**
Related to p1710864734866069-slack-C0347E545HR

## Proposed Changes

This PR changes the `Learn more` links on `Hosting Configuration` page from:

* `Learn more about SFTP on WordPress.com`: from https://developer.wordpress.com/docs/developer-tools/sftp-ssh/ to https://developer.wordpress.com/docs/developer-tools/sftp/
* `Learn more` by the `Enable SSH access for this site` on Hosting Configuration card: from https://wordpress.com/support/connect-to-ssh-on-wordpress-com/ to https://developer.wordpress.com/docs/developer-tools/ssh/
* `Learn more` by `Manage your site’s server-side caching`: from https://wordpress.com/support/clear-your-sites-cache/ to https://developer.wordpress.com/docs/site-performance/global-edge-cache/#1-clear-your-site-s-cache

## Testing Instructions

* Navigate to Calypso live link generated by this branch
* Make sure you have an Atomic site
* Navigate to `Settings > Hosting Configuration`: /hosting-config/{yourAtomicSite}
* Look for `Learn more about SFTP on WordPress.com` on the top of the SFTP/SSH credentials card
* Confirm that clicking the link opens the inline help article from https://developer.wordpress.com/docs/developer-tools/sftp/
* Navigate through an arrow from inline help and confirm you are brought to the right page:

Screenshot 2024-03-21 at 4 14 48 PM

* Look for the `Learn more` link to the `Enable SSH access for this site` toggle
* Confirm that clicking the link opens the inline help article from https://developer.wordpress.com/docs/developer-tools/ssh/
* Navigate through an arrow from inline help and confirm you are brought to the right page:

Screenshot 2024-03-21 at 4 16 19 PM

* `Learn more` by `Manage your site’s server-side caching`
* Confirm that clicking the link opens the inline help article from https://developer.wordpress.com/docs/site-performance/global-edge-cache/#1-clear-your-site-s-cache
* Navigate through an arrow from inline help and confirm you are brought to the right page:

Screenshot 2024-03-21 at 4 17 32 PM

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by alexapeduzzi** [comment]
It looks good to me, @katinthehatsite! You're the best. Thanks!

File: client/components/inline-support-link/context-links.js [hunk 1]
```diff
@@ -106,11 +106,11 @@ const contextLinks = {
 		blog_id: DEVELOPER_WORDPRESS_BLOG_ID,
 	},
 	'hosting-clear-cache': {
-		link: 'https://wordpress.com/support/clear-your-sites-cache/',
+		link: 'https://developer.wordpress.com/docs/site-performance/global-edge-cache/#1-clear-your-site-s-cache',
 		post_id: 164969,
```
Comment (sejas) [inline client/components/inline-support-link/context-links.js]: @katinthehatsite , should we add the `blog_id: DEVELOPER_WORDPRESS_BLOG_ID,` and change the `post_id` to `99415`?

I didn't check the code, so not sure why are those parameters required. My best guess is for analytics?

File: client/components/inline-support-link/context-links.js [hunk 2]
```diff
@@ -106,11 +106,11 @@ const contextLinks = {
...
 	},
 	'hosting-connect-to-ssh': {
-		link: 'https://wordpress.com/support/connect-to-ssh-on-wordpress-com/',
+		link: 'https://developer.wordpress.com/docs/developer-tools/ssh/',
 		post_id: 213309,
```
Comment (sejas) [inline client/components/inline-support-link/context-links.js]: Similar to the previous one. We may want to add `blog_id: DEVELOPER_WORDPRESS_BLOG_ID,` and change the post_id to `100385`.

**Review by sejas — APPROVED** [review]
@katinthehatsite , thanks for stepping in for this change.

I tested the three links and they work as expected.
I just have one question related to `post_id` and `blog_id`.
Do you know why they are necessary ? Should we update them for consistency?

File: client/components/inline-support-link/context-links.js (hunk 1 above)
Comment (katinthehatsite) [reply client/components/inline-support-link/context-links.js]: Yes, we definitely should! Nice catch!

I updated it in https://github.com/Automattic/wp-calypso/pull/88766/commits/58092cc6df9a77d21aec91506dd51096916fbc1a

Thanks for mentioning this!

File: client/components/inline-support-link/context-links.js (hunk 2 above)
Comment (katinthehatsite) [reply client/components/inline-support-link/context-links.js]: Done in https://github.com/Automattic/wp-calypso/pull/88766/commits/58092cc6df9a77d21aec91506dd51096916fbc1a

File: client/components/inline-support-link/context-links.js (hunk 1 above)
Comment (katinthehatsite) [reply client/components/inline-support-link/context-links.js]: @sejas I checked why the `post_id` is there and apparently it is the following:

`If props.supportPostId is supplied, when the link is clicked a SupportArticleDialog is opened with an ExternalLink to the props.supportLink. If no props.supportPostId is supplied, an ExternalLink to the props.supportLink is rendered`

I found this in the `InlineSupportLink` README: https://github.com/Automattic/wp-calypso/blob/d4616fb44fbd86a46f10a6db2d6f145aaadf73da/client/components/inline-support-link/README.md

So basically if you are not providing the post_id, then the link will open directly in the browser. If you are providing the post id, the link will render the help dialog with the article.

**Comment by wojtekn** [comment]
Thanks for making this change @katinthehatsite , looks great!
